### ISSUE TITLE
exp/shiny: key repeats reported as release + press

### ISSUE THREAD

**Issue by barnex** [golang/go#18566]
### What did you do?

This program opens an x/exp/shiny window and reports key presses:

```
package main

import (
    "log"
    "golang.org/x/exp/shiny/driver"
    "golang.org/x/exp/shiny/screen"
    "golang.org/x/mobile/event/key"
)

func main() {
    log.SetFlags(log.Lmicroseconds)
    driver.Main(func(s screen.Screen) {
        w, _ := s.NewWindow(nil)
        for {
            e := w.NextEvent()
            if e, ok := e.(key.Event); ok {
                log.Println(e.Direction)
            }
        }
    })
}
```

### What did you expect to see?
According to https://github.com/golang/mobile/blob/master/event/key/key.go#L50, key repeats should report key.DirNone. So when pressing and holding down a key for a while, I expect to see:
```
11:59:49.544507 Press
11:59:50.045002 None
11:59:50.076717 None
11:59:50.107537 None
11:59:50.137585 Release
```

### What did you see instead?
Holding down a key results in repeated Release/Press events:

```
11:59:49.544507 Press
11:59:50.045002 Release
11:59:50.045110 Press
11:59:50.076717 Release
11:59:50.076807 Press
11:59:50.107537 Release
11:59:50.107578 Press
11:59:50.137585 Release
```

### What version of Go are you using (`go version`)?
`go version go1.7.1 linux/amd64`

### What operating system and processor architecture are you using (`go env`)?
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/arne"
GORACE=""
GOROOT="/home/arne/bin/go"
GOTOOLDIR="/home/arne/bin/go/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build871086202=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
```

**Comment by as** [comment]
I wasn't able to reproduce this on NT, so I looked to Linux as the culprit.

According to [xorg](https://x.org/releases/X11R7.7/doc/kbproto/xkbproto.html#Detectable_Autorepeat)

> The X server usually generates both press and release events whenever an autorepeating key is held down. If an XKB-aware client enables the DetectableAutorepeat per-client option for a keyboard, the server sends that client a key release event only when the key is physically released. For example, holding down a key to generate three characters without detectable autorepeat yields:

>Press -> Release -> Press -> Release -> Press -> Release
>If detectable autorepeat is enabled, the client instead receives:

>Press-> Press -> Press -> Release
>Note that only clients that request detectable autorepeat are affected; other clients continue to receive both press and release events for autorepeating keys. Also note that support for detectable autorepeat is optional; servers are not required to support detectable autorepeat, but they must correctly report whether or not it is supported.

**Comment by nigeltao** [comment]
Yeah, it's a bug in the x11driver, although if fixing this involves XKB's DetectableAutorepeat, as @as suggested, then it's not going to be trivial to fix. The xgb library that we use says, in https://github.com/BurntSushi/xgb/blob/master/doc.go: "XKB is the only extension that intentionally does not work... due to its complexity and my current mental incapacity to test it."
